test/custom_task_test.go

```diff
@@ -295,36 +330,47 @@ spec:
 	}
 
 	// Get the Run name.
-	if len(pr.Status.Runs) != 1 {
-		t.Fatalf("PipelineRun had unexpected .status.runs; got %d, want 1", len(pr.Status.Runs))
-	}
+	runName := ""
 
-	for runName := range pr.Status.Runs {
-		// Get the Run.
-		r, err := c.RunClient.Get(ctx, runName, metav1.GetOptions{})
-		if err != nil {
-			t.Fatalf("Failed to get Run %q: %v", runName, err)
+	if embeddedStatusValue != config.MinimalEmbeddedStatus {
+		if len(pr.Status.Runs) != 1 {
+			t.Fatalf("PipelineRun had unexpected .status.runs; got %d, want 1", len(pr.Status.Runs))
 		}
-		if r.IsDone() {
-			t.Fatalf("Run unexpectedly done: %v", r.Status.GetCondition(apis.ConditionSucceeded))
+		for rn := range pr.Status.Runs {
```

Comment:
I'm a bit confused, what's going on here?

Reply:
We need the run name later in the test.

Reply:
oh I see, the slice has length 1-- in that case pr.Status.Runs[0] is more clear

Reply:
Ah, right, I tried to do that, but `pr.Status.Runs` is a map, so `pr.Status.Runs[0]` isn't an option.